client/dashboard/data/me-active-subscriptions.ts

```diff
@@ -0,0 +1,188 @@
+	currency_code: string;
+	currency_symbol: string;
+	description: string;
+	domain: string;
+	domain_registration_agreement_url: string | undefined;
+	blog_created_date: string;
+	expiry_date: string;
+	expiry_status: string;
+	iap_purchase_management_link: string | null;
+	included_domain: string;
+	included_domain_purchase_amount: number;
+	introductory_offer: RawPurchaseIntroductoryOffer | null;
+	is_cancelable: boolean;
+	is_domain: boolean;
+	is_domain_registration: boolean;
+	is_free_jetpack_stats_product: boolean;
+	is_google_workspace_product: boolean;
+	is_hundred_year_domain: boolean;
+	is_iap_purchase: boolean;
+	is_jetpack_ai_product: boolean;
+	is_jetpack_stats_product: boolean;
+	is_locked: boolean;
+	is_plan: boolean;
+	is_rechargable: boolean;
+	is_refundable: boolean;
+	is_renewable: boolean;
+	is_renewal: boolean;
+	is_titan_mail_product: boolean;
+	is_woo_express_trial: boolean;
+	meta: string | undefined;
+	ownership_id: number | undefined;
+	partner_name: string | undefined;
+	partner_slug: string | undefined;
+	partner_type: string | undefined;
+	partner_key_id: number | undefined;
+	payment_name: string;
+	payment_type:
+		| 'credit_card'
+		| 'paypal_direct'
+		| 'paypal'
+		| 'emergent-paywall'
+		| 'brazil-tef'
+		| string;
+	payment_card_display_brand: string | null;
+	payment_country_name: string;
+	payment_country_code: string | null;
+	stored_details_id: string | null;
+	pending_transfer: boolean;
+	product_id: number | string;
+	product_name: string;
+	product_slug: string;
+	product_type: string;
+	product_display_price: string;
+	price_integer: number;
+	purchaser_id?: number;
+	total_refund_amount: number | undefined;
+	total_refund_currency: string;
+	total_refund_integer: number;
+	total_refund_text: string;
+	refund_amount: number;
+	refund_integer: number;
+	refund_text: string;
+	refund_currency_symbol: string;
+	refund_options: RefundOptions | null;
+	refund_period_in_days: number;
+	regular_price_text: string;
+	regular_price_integer: number;
+	renew_date: string;
+	sale_amount: number | undefined;
+	sale_amount_integer: number | undefined;
+	blog_id: number | string;
+	blogname: string;
+	site_slug?: string;
+	subscribed_date: string;
+	subscription_status: 'active' | 'inactive';
+	tag_line: string;
+	tax_amount: number | string | undefined;
+	tax_text: string | undefined;
+	renewal_price_tier_usage_quantity: number | undefined | null;
+	user_id: number | string;
+	auto_renew: '1' | '0' | null;
+	payment_card_id: number | string | undefined;
+	payment_card_type: string | undefined;
+	payment_card_processor: string | undefined;
+	payment_details: string | undefined;
+	payment_expiry: string | undefined;
+}
+
+export async function fetchActiveSubscriptionsForUser( options?: {
+	siteId?: string | number;
+} ): Promise< ActiveSubscription[] > {
+	if ( options?.siteId ) {
+		return await wpcom.req.get( {
+			path: `/sites/${ encodeURIComponent( options.siteId ) }/purchases`,
```

Comment:
Team review: [nit] Since the fetch function uses different functions, consider create separate fetch functions.